lib/sth_server.js

```diff
@@ -768,6 +768,99 @@ function startServer(host, port, aSTHDatabase, callback) {
     }
   }
 
+  /**
+   * Handler in case a removal request is received
+   * @param request The received request
+   * @param reply hapi's server reply() function
+   */
+  function removeDataHandler(request, reply) {
+    request.sth = request.sth || {};
+    request.sth.context = getContext(request);
+
+    sthLogger.debug(
+      request.sth.context,
+      request.method.toUpperCase() + ' ' + request.url.path +
+        ' with headers: ' + JSON.stringify(request.headers)
+    );
+
+    sthDatabase.removeData(
+      {
+        service: request.headers['fiware-service'],
+        servicePath: request.headers['fiware-servicepath'],
+        entityId: request.params && request.params.entityId,
+        entityType: request.params && request.params.entityType,
+        attrName: request.params && request.params.attrName
+      },
+      function(err) {
+        if (err) {
+          if (err.name === 'MongoError' && err.message.indexOf('does not exist. Currently in strict mode')) {
+            // There is no associated data for the provided service, service path and entity
+            sthLogger.debug(
+              request.sth.context,
+              'No data associated to the provided service (' + request.headers['fiware-service'] +
+                '), service path (' + request.headers['fiware-servicepath'] + ')' +
+                ', entity id (' + request.params.entityId + ')' +
+                ', entity type (' + request.params.entityType + ')' +
+                (request.params.attrName ? ', attribute name (' + request.params.attrName : '') +
+                 ' available'
+            );
+            // Reply with no error
+            return reply();
+          } else {
+            sthLogger.warn(
+              request.sth.context,
+              'Error when removing the data associated to an entity: ' + err
+            );
+            // Reply with error
+            return reply(err);
+          }
+        }
+        sthLogger.debug(
+          request.sth.context,
+          'Data associated to the provided service (' + request.headers['fiware-service'] +
+            '), service path (' + request.headers['fiware-servicepath'] + ')' +
```

Comment:
Maybe you could use a function that generates the string identifying the service, for using in debug logs, in order to use make logs look like:
```
sthLogger.debug(request.sth.context, 'No data associated to the provided '  + getIdentificationTest (request) + ' available' );
```